lib/matplotlib/colorbar.py

```diff
@@ -900,6 +900,406 @@ def remove(self):
+            locator.set_view_interval(*intv)
+            locator.set_data_interval(*intv)
+            formatter.set_view_interval(*intv)
+            formatter.set_data_interval(*intv)
+
+            b = np.array(locator())
+            if isinstance(locator, ticker.LogLocator):
+                eps = 1e-10
+                b = b[(b <= intv[1] * (1 + eps)) & (b >= intv[0] * (1 - eps))]
+            else:
+                eps = (intv[1] - intv[0]) * 1e-10
+                b = b[(b <= intv[1] + eps) & (b >= intv[0] - eps)]
+            # self._tick_data_values = b
+            ticks = self._locate(b, norm)
+            formatter.set_locs(b)
+            ticklabels = [formatter(t, i) for i, t in enumerate(b)]
+            offset_string = formatter.get_offset()
+            return ticks, ticklabels, offset_string
+
+        ax = self.ax
+        xticks, xticklabels, xoffset_string = _make_ticker(self.norm.norm1)
+        yticks, yticklabels, yoffset_string = _make_ticker(self.norm.norm2)
+
+        ax.xaxis.set_ticks(xticks)
+        ax.set_xticklabels(xticklabels)
+        ax.xaxis.get_major_formatter().set_offset_string(xoffset_string)
+
+        ax.yaxis.set_ticks(yticks)
+        ax.set_yticklabels(yticklabels)
+        ax.yaxis.get_major_formatter().set_offset_string(yoffset_string)
+
+    def set_ticks(self, xticks, yticks, update_ticks=True):
+        if cbook.iterable(xticks):
+            self.xlocator = ticker.FixedLocator(xticks, nbins=len(xticks))
+        else:
+            self.xlocator = xticks
+
+        if cbook.iterable(yticks):
+            self.ylocator = ticker.FixedLocator(yticks, nbins=len(yticks))
+        else:
+            self.ylocator = yticks
+
+        if update_ticks:
+            self.update_ticks()
+        self.stale = True
+
+    def set_ticklabels(self, xticklabels=None, yticklabels=None,
+                       update_ticks=True):
+        """
+        set tick labels. Tick labels are updated immediately unless
+        update_ticks is *False*. To manually update the ticks, call
+        *update_ticks* method explicitly.
+        """
+        if xticklabels is not None or yticklabels is not None:
+            if isinstance(self.xlocator, ticker.FixedLocator):
+                self.xformatter = ticker.FixedFormatter(xticklabels)
+
+            if isinstance(self.ylocator, ticker.FixedLocator):
+                self.yformatter = ticker.FixedFormatter(yticklabels)
+
+            if update_ticks:
+                self.update_ticks()
+
+            self.stale = True
+            return
+
+        warnings.warn("set_ticks() must have been called.")
+        self.stale = True
+
+    def _set_label(self):
+        self.ax.set_ylabel(self._ylabel, **self._labelkw)
+        self.ax.set_xlabel(self._xlabel, **self._labelkw)
+        self.stale = True
+
+    def set_label(self, xlabel, ylabel, **kw):
+        """
+        Label the axes of the colorbar
+        """
+        self._xlabel = '%s' % (xlabel, )
```

Comment:
just self._xlabel = xlabel should be enough (unless we explicitly want to support any object via conversion to string, and even then str(xlabel) is clearer.